src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java

```diff
@@ -184,25 +185,32 @@ public static String getCollection(Firestore database, String collectionToGetFro
    *   if null, then execute getFirstBatch().
    * @param parameterToSortBy - The parameter I want to presort my collection by, 
    *   if null then don't sort.
+   * @param sortOrder - The sort order (true for ascending and false for descending).
    * @return - The json string representing the new batch of documents to be sent to the frontend.
    * @throws FirebaseException
    */
   public static String getCollectionBatch(Firestore database, String collectionToGetFrom, 
-      Optional<Integer> batchSizeLimit,  Optional<String> lastDocID, Optional<String> parameterToSortBy) 
-      throws FirebaseException {
+      Optional<Integer> batchSizeLimit,  Optional<String> lastDocID, 
+      Optional<String> parameterToSortBy, Optional<Boolean> sortOrder) throws FirebaseException {
     // A batch size limit needs to be specified in order to make any query.
     if (!batchSizeLimit.isPresent() || batchSizeLimit.get() == 0) {
       throw new FirebaseException(BATCH_SIZE_NOT_SPECIFIED);
     }
 
     CollectionReference collectionReference = database.collection(collectionToGetFrom);
 
+    Direction sortDirection = sortOrder.isPresent() 
```

Comment:
Why not just do a sortOrder.get() ? ... : ...;

Reply:
sortOrder.get() will throw an error if there's a null in Optional. In this case I want to handle the null case by sorting by default order (descending)

Reply:
Is there a better way to handle both the null case and the false case? quq

Reply:
You could create a boolean that checks if the sortOrder is null or false. eg "bool falseSortOrder = sortOrder.present() == null || !sortOrder.get();' and then do the check on that eg "falseSortOrder ? descending : ascending;"

Reply:
Ah yeah I can use orElse! Thanks for the reminder, it is much cleaner now.